src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/PrecompiledRazorPageSuppressor.cs

```diff
@@ -65,12 +66,8 @@ private void Workspace_WorkspaceChanged(object sender, WorkspaceChangeEventArgs
                         break;
                     }
 
-                    if (!document.FilePath.Contains("RazorDeclaration"))
-                    {
-                        // Razor output file that is not a declaration file.
-                        _workspace.RemoveDocument(document.Id);
-                        break;
-                    }
+                    // Razor output file
+                    _workspace.RemoveDocument(document.Id);
```

Comment:
Can the `if` block in line 51 now be removed?

Reply:
Sadly not because those two cases would not result in the document being removed due to those files not meeting the following condition

```C#
if (!document.FilePath.EndsWith(".cshtml.g.cs", StringComparison.Ordinal) && 
    !document.FilePath.EndsWith(".razor.g.cs", StringComparison.Ordinal) &&

    // 2.2 only extension for generated Razor files
    !document.FilePath.EndsWith("g.cshtml.cs", StringComparison.Ordinal))
{
    break;
}
```